### PR TITLE
Upgrade to 1.4.0

### PR DESCRIPTION
**Pull request by knikos**
## Why this should be merged
After banff16, e2e tests in camino-node fail. Since they run against the network-runner, an upgrade is necessary.
Required version according to avalanchego is v1.3.5-rc.0 (see https://github.com/ava-labs/avalanchego/blob/v1.9.16/scripts/tests.e2e.sh#L46). However, interdependencies with caminogo require version v1.4.0. 

## How this works
Merges changes of network-runner up to  https://github.com/ava-labs/avalanche-network-runner/releases/tag/v1.4.0 and resolve conflicts.

## How this was tested
Automatic unit tests. E2e tests should be ignored until new node version is available.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -141,6 +123,6 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-replace github.com/ava-labs/avalanchego => github.com/chain4travel/caminogo v0.4.1
+replace github.com/ava-labs/avalanchego => github.com/chain4travel/caminogo v0.4.9-rc2.0.20230526125621-0adfa80bd7c4
```
Comment (evlekht) [inline go.mod:126]: why not use our banff16 caminogo and caminoevm commits?

File: tests/e2e/e2e_test.go [hunk 2]
```diff
@@ -423,65 +430,216 @@ var _ = ginkgo.Describe("[Start/Remove/Restart/Add/Stop]", func() {
 			gomega.Ω(err).Should(gomega.BeNil())
 		})
 	})
+	ginkgo.It("can pause a node", func() {
```
Comment (evlekht) [inline tests/e2e/e2e_test.go:433]: in diff with avax file we still have bunch of test cases missing, is this ok?

Comment (knikos) [reply tests/e2e/e2e_test.go:433]: these tests were failing and I removed cause they had been also removed from an older avax merge:
see here
https://github.com/chain4travel/camino-network-runner/compare/6d39aae6022bc8a89d3f8cdcea8324a57d05755b...de4c2795e3d1e3443a5bd765413d43523aaa6721#diff-7d1fc1e232b55677582b51e2623aae6d8c15b5f0960ac6100fc7d733cb795dacL149
I asked about the reasoning behind this removal, but the colleagues cannot remember.

File: go.mod (hunk 1 above)
Comment (knikos) [reply go.mod:126]: will update as soon as the new node version is also released. We can tag the commits on these repos as well.

**Review by c4t-ag — APPROVED** [review]

**Pull request merged**